### PR TITLE
OHRI-1172 Latest pregnancy information displayed

### PR DESCRIPTION
**Pull request by kajambiya**

### CODE REVIEW

**Comment by samuelmale** [comment]
@kajambiya did you see the build error?

File: packages/esm-ohri-pmtct-app/src/api/api.ts [hunk 1]
```diff
@@ -43,3 +45,24 @@ export function ancVisitsReportCount(pTrackerID: string, patientUuid: string) {
     `/ws/rest/v1/reportingrest/dataSet/${ancVisitsReport}?patient_uuid=${patientUuid}&ptracker_id=${pTrackerID}`,
   );
 }
+
+//fetch current Ptraxker
```
Comment (samuelmale) [inline packages/esm-ohri-pmtct-app/src/api/api.ts:49]: Please remove this comment.

File: packages/esm-ohri-pmtct-app/src/api/api.ts [hunk 2]
```diff
@@ -43,3 +45,24 @@ export function ancVisitsReportCount(pTrackerID: string, patientUuid: string) {
...
+export function fetchPatientIdentifiers(patientUuid: string) {
+  return openmrsFetch(`${BASE_WS_API_URL}/patient/${patientUuid}/identifier`).then(({ data }) => {
+    if (data.results.length) {
+      return data.results;
+    }
+    return null;
+  });
+}
+
+//Save current PTracker
```
Comment (samuelmale) [inline packages/esm-ohri-pmtct-app/src/api/api.ts:59]: Remove comment

File: packages/esm-ohri-pmtct-app/src/form-entry/post-submission-actions/mother-child-linkage-action.ts [hunk 3]
```diff
@@ -87,6 +96,24 @@ async function constructPatientObjectFromObsData(obsGroup, encounterLocation: st
...
+
+  //add current ptracker to identities
+  const currentPTrackerObject: PatientIdentifier = {
+    identifier: inComingPTrackerID,
+    identifierType: PTrackerIdentifierType,
+    location: encounterLocation,
+    preferred: false,
+  };
+  saveIdentifier(currentPTrackerObject, patientUuid);
+}
```
Comment (samuelmale) [inline packages/esm-ohri-pmtct-app/src/form-entry/post-submission-actions/mother-child-linkage-action.ts:116]: Can you re-use this code for both submission handlers? You have code duplication.

**Review by samuelmale — CHANGES_REQUESTED** [review]

**Comment by kajambiya** [comment]
> @kajambiya did you see the build error?

Yah, hadn't rebased on latest dev but have pushed a commit

File: packages/esm-ohri-pmtct-app/src/form-entry/post-submission-actions/mother-child-linkage-action.ts (hunk 3 above)
Comment (kajambiya) [reply packages/esm-ohri-pmtct-app/src/form-entry/post-submission-actions/mother-child-linkage-action.ts:116]: Yah, hadn't rebased on latest dev but have pushed a commit

File: packages/esm-ohri-pmtct-app/src/api/api.ts (hunk 1 above)
Comment (kajambiya) [reply packages/esm-ohri-pmtct-app/src/api/api.ts:49]: oh,ok

File: packages/esm-ohri-pmtct-app/src/form-entry/post-submission-actions/mother-child-linkage-action.ts (hunk 3 above)
Comment (kajambiya) [reply packages/esm-ohri-pmtct-app/src/form-entry/post-submission-actions/mother-child-linkage-action.ts:116]: ok

**Comment by kajambiya** [comment]
@samuelmale I've addressed the suggestions as per your comments. Kindly confirm

File: packages/esm-ohri-pmtct-app/src/form-entry/post-submission-actions/current-ptracker-action.ts [hunk 4]
```diff
@@ -7,32 +7,33 @@ export const PTrackerSubmissionAction: PostSubmissionAction = {
   applyAction: async function ({ patient, encounters, sessionMode }) {
     const encounter = encounters[0];
     const encounterLocation = encounter.location['uuid'];
+    const patientUuid = patient['id'];
 
     // Don't do post submission if action is view
     if (sessionMode === 'view') {
       return;
     }
+    updatePatientPtracker(encounter, encounterLocation, patientUuid);
```
Comment (samuelmale) [inline packages/esm-ohri-pmtct-app/src/form-entry/post-submission-actions/current-ptracker-action.ts:16]:
```js

updatePatientPtracker(encounter, encounterLocation, patient.id);

```

File: packages/esm-ohri-pmtct-app/src/form-entry/post-submission-actions/current-ptracker-action.ts [hunk 5]
```diff
@@ -7,32 +7,33 @@ export const PTrackerSubmissionAction: PostSubmissionAction = {
   applyAction: async function ({ patient, encounters, sessionMode }) {
     const encounter = encounters[0];
     const encounterLocation = encounter.location['uuid'];
+    const patientUuid = patient['id'];
```
Comment (samuelmale) [inline packages/esm-ohri-pmtct-app/src/form-entry/post-submission-actions/current-ptracker-action.ts:10]: You don't need this.

**Review by samuelmale — DISMISSED** [review]
LGTM 🍺 

Just a few minor comments..

File: packages/esm-ohri-pmtct-app/src/form-entry/post-submission-actions/current-ptracker-action.ts (hunk 5 above)
Comment (kajambiya) [reply packages/esm-ohri-pmtct-app/src/form-entry/post-submission-actions/current-ptracker-action.ts]: Done

**Review by samuelmale — APPROVED** [review]

**Pull request merged**